### PR TITLE
Add ADR-0022 - Integration Quality Scale

### PR DESCRIPTION
**Pull request by joostlek**
Add ADR-0022 - Integration Quality Scale

### CODE REVIEW

File: adr/0022-integration-quality-scale.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+This scale provides a framework for everybody contributing to enhancing the integrations while offering transparency regarding the reliability and capabilities of integrations offered to our users.
+
+Here is a quick overview of the different tiers in the quality integration scale:
+
+🥉 Bronze: This is the baseline standard and requirement for all new integrations added to Home Assistant. The integration can be configured via the UI, and there is documentation on how to set it up.
+
+🥈 Silver: Enhanced reliability and robustness of the integration, ensuring a solid runtime experience. The integration handles errors correctly, including authentication errors and offline devices. The documentation provides more information on what the integration exactly provides. The integration has one or more code owners who help to maintain the integration and ensure this runtime experience lasts.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:19]:
```suggestion
🥈 Silver: Enhanced reliability and robustness of the integration, ensuring a solid runtime experience. The integration handles errors correctly, including authentication errors and offline devices. The documentation provides more information on what the integration provides. The integration has one or more code owners who help to maintain the integration and ensure this runtime experience lasts.
```

File: adr/0022-integration-quality-scale.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+🥇 Gold: The gold standard in delivering the best possible user experience with advanced features and comprehensive support. They are fully featured, user-friendly, and accessible to a wider audience. It is fully translated, capable of discovering their devices, providing device updates through Home Assistant, and has extensive documentation with examples, use cases, and troubleshooting. They provide means for debugging issues and can be fully adjusted through the UI.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:21]:
```suggestion
🥇 Gold: The gold standard in delivering the best possible user experience with advanced features and comprehensive support. The integration is fully featured, user-friendly, and accessible to a wider audience. It is fully translated, capable of discovering its devices, providing device updates through Home Assistant, and has extensive documentation with examples, use cases, and troubleshooting. The integration provides means for debugging issues and can be fully adjusted through the UI.
```

File: adr/0022-integration-quality-scale.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+🏆 Platinum: The epitome of quality within Home Assistant. Not only does this integration provide an outstanding user experience, but it also adheres to the highest technical standards with optimized performance, efficiency, and code quality.
+
+Besides the four scaled tiers above, we also have four additional special tiers:
+
+❓ No score: An integration that is available via the UI but has not been evaluated against the integration quality scale or does not meet the minimum requirements for scoring.
+
+🏠 Internal: This is an integration used internally by Home Assistant to provide basic components and building blocks for the core program of Home Assistant or for other integrations to built on. The Home Assistant project maintains these.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:29]:
```suggestion
🏠 Internal: This is an integration used internally by Home Assistant to provide basic components and building blocks for the core program of Home Assistant or for other integrations to build on. The Home Assistant project maintains integrations of this level.
```

File: adr/0022-integration-quality-scale.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+💾 Legacy: An older integration that has not yet transitioned to the Home Assistant user interface and lacks modern (nowadays considered normal) features.
+
+📦 Custom: The Home Assistant project does not create, maintain, audit, review, or support custom integrations; therefore, they cannot be graded and placed on the integration quality scale.
+
+## Decision
+
+### Scaled tiers
+
+Home Assistant integration quality scale has four differentiated tiers to which integration can be graded: 🥉 Bronze, 🥈 Silver, 🥇 Gold, and 🏆 Platinum.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:39]:
```suggestion
Home Assistant integration quality scale has four differentiated tiers to which an integration can be graded: 🥉 Bronze, 🥈 Silver, 🥇 Gold, and 🏆 Platinum.
```

File: adr/0022-integration-quality-scale.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+Each tier builds on the previous tier; for example, “Silver” has everything required by “Bronze”, and “Gold” has everything both “Silver” and “Bronze” require.
+
+While some integrations may not be graded in the top tier, they might still implement parts of other (higher grade) tiers.
+
+#### 🥉 Bronze
+The bronze tier is the baseline standard and requirement for all new integrations. It meets the minimum requirements in code quality, functionality, and user experience. It complies with the fundamental expectations and provides a reliable foundation for users to interact with their devices and services.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:46]: We don't need to repeat the description of the tiers again. Just reference the above text, then simply list the named rules required for each tier.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]
Try to shorten the text down. It's too much repetition and essay-style.

File: adr/0022-integration-quality-scale.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+In order for these devices to be able to get certified, the integration used to integrate the device or service must have reached at least the “Gold” tier. The “Gold” tier is specifically designed to provide the best possible user experience possible.
+
+If rules to our quality integration scales have been adjusted, it is possible that an integration with the Works with Home Assistant program doesn’t comply with the newly adjusted standards yet. At that moment, the existing “Gold” tier is grandfathered. Once a new or extra device is certified for the same integration, the integration must be updated to also comply with the new or adjusted standards.
+
+When a device (or devices) are all removed from integration, the grandfathered flag will also be released and the integration quality scale tier adjusted to the current status of the integration.
+
+## Consequences
+
+- A clear and transparent framework for contributors to understand the quality expectations for their integrations.
+- An increased pressure on developer documentation to make sure it is up-to-date.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:169]:
```suggestion
- An increased emphasis on developer documentation to make sure it is up-to-date.
```

File: adr/0022-integration-quality-scale.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+In order for these devices to be able to get certified, the integration used to integrate the device or service must have reached at least the “Gold” tier. The “Gold” tier is specifically designed to provide the best possible user experience possible.
+
+If rules to our quality integration scales have been adjusted, it is possible that an integration with the Works with Home Assistant program doesn’t comply with the newly adjusted standards yet. At that moment, the existing “Gold” tier is grandfathered. Once a new or extra device is certified for the same integration, the integration must be updated to also comply with the new or adjusted standards.
+
+When a device (or devices) are all removed from integration, the grandfathered flag will also be released and the integration quality scale tier adjusted to the current status of the integration.
+
+## Consequences
+
+- A clear and transparent framework for contributors to understand the quality expectations for their integrations.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:168]:
```suggestion
The integration quality scale will:
- give a clear and transparent framework for contributors to understand the quality expectations for their integrations.
```

File: adr/0022-integration-quality-scale.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+- An increased emphasis on developer documentation to make sure it is up-to-date.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:169]:
```suggestion
- put pressure on the maintenance of the developer documentation to make sure the documentation is up-to-date.
```

File: adr/0022-integration-quality-scale.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+Accepted
+
+## Context
+
+Home Assistant is a powerful open-source home automation system that enables you and all other users to automate and control various smart devices and services in their homes. We have established the integration quality scale to ensure a high-quality and consistent user and contributor experience for our open-source project.
+
+This scale provides a framework for everybody contributing to enhancing the integrations while offering transparency regarding the reliability and capabilities of integrations offered to our users.
+
+Here is a quick overview of the different tiers in the quality integration scale:
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:15]:
```suggestion
Here is an overview of the different tiers in the quality integration scale:
```

File: adr/0022-integration-quality-scale.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+Characteristics:
+- Not yet scored or lacks sufficient information for scoring.
+- Can be set up via the UI, but may need enhancements for a better experience.
+- May function correctly, but hasn’t been verified against current standards.
+- Documentation most often provides only basic setup steps.
+
+#### 🏠 Internal
+This tier is assigned to integrations used internally by Home Assistant. These integrations provide basic components and building blocks for Home Assistant's core program or for other integrations to build on top of it.
+
+Everything in Home Assistant is extendable using integrations. In fact, Home Assistant uses its own integration logic to extend base functionality. For example, the automation engine of Home Assistant is an integration. Those internal parts are thus marked with this “Internal” tier.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:105]: I'd remove this paragraph. It'd not needed. If wanted put the example in the first paragraph of this section instead.

File: adr/0022-integration-quality-scale.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+The gold tier has the following characteristics:
+- Provides everything “Silver” has.
+- Has the best end-user experience an integration can offer; streamlined and intuitive.
+- Can be automatically discovered, simplifying the integration setup.
+- Integration can be reconfigured and adjusted.
+- Supports translations.
+- Extensive documentation, aimed at non-technical users.
+- It supports updating the software/firmware of devices through Home Assistant when possible.
+- Provides the tooling and documentation troubleshooting issues.
```
[the remark this answers is not in the document]
Comment (joostlek) [reply adr/0022-integration-quality-scale.md:73]:
```suggestion
```

File: adr/0022-integration-quality-scale.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+
+Once the Home Assistant core team reviews and approves it, the integration will display the new tier as of the next major release of Home Assistant.
+
+Besides upgrading an integration to a higher tier on the scale, it is also possible for an integration to be downgraded to a lower tier. This can, for example, happen when there is no longer an active integration code owner. In this specific example, the integration will be downgraded to “Bronze,” even if it otherwise fully complies with the “Platinum” tier.
+
+### Adjustments to rules contained in each tier
+The world of IoT and all technologies used by Home Assistant are changing at a fast pace; not just in terms of what Home Assistant can support or do, but also in terms of the software on which Home Assistant is built, which is constantly innovating. Home Assistant is pioneering the technology in the industry at a fast pace.
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:147]: We already say it's changing fast.
```suggestion
The world of IoT and all technologies used by Home Assistant are changing at a fast pace; not just in terms of what Home Assistant can support or do, but also in terms of the software on which Home Assistant is built. Home Assistant is pioneering the technology in the industry at a fast pace.
```

**Review by MartinHjelmare — APPROVED** [review]

File: adr/0022-integration-quality-scale.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
+# 22. Integration quality scale
+
+Date: yyyy-mm-dd
```
Comment (MartinHjelmare) [inline adr/0022-integration-quality-scale.md:3]: Remember to update this before merging.

File: adr/0022-integration-quality-scale.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+- A fully asynchronous integration code base ensures efficient operation.
+- Implements efficient data handling, reducing network and CPU usage.
+
+### Special tiers
+
+Besides the four scaled tiers mentioned above, Home Assistant also provides four special tiers to denote a special case, causing them not to fit into the regular scale set out above.
+
+Note that while an integration might have been given one of these special tiers, they still might pursue the integration quality equivalent to any of the other scaled tiers.
+
+#### ❓ No score
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:90]:
```suggestion
#### ❓ No score

```

File: adr/0022-integration-quality-scale.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+
+The “No score” tier cannot be assigned to new integrations, as they are required to have at least a “Bronze” level when introduced. The Home Assistant project encourages the community to help update these integrations without a score to meet at least the “Bronze” level requirements.
+
+Characteristics:
+- Not yet scored or lacks sufficient information for scoring.
+- Can be set up via the UI, but may need enhancements for a better experience.
+- May function correctly, but hasn’t been verified against current standards.
+- Documentation most often provides only basic setup steps.
+
+#### 🏠 Internal
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:101]:
```suggestion
#### 🏠 Internal

```

File: adr/0022-integration-quality-scale.md [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+This tier is assigned to integrations used internally by Home Assistant. These integrations provide basic components and building blocks for Home Assistant's core program or for other integrations to build on top of it.
+
+Internal integrations are maintained by the Home Assistant project and subjected to strict architectural design procedures.
+
+Characteristics:
+- Internal, built-in building blocks of the Home Assistant core program.
+- Provides building blocks for other integrations to use and build on top of.
+- Maintained by the Home Assistant project.
+
+#### 💾 Legacy
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:111]:
```suggestion
#### 💾 Legacy

```

File: adr/0022-integration-quality-scale.md [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+
+The Home Assistant project encourages the community to help migrate these integrations to the UI and update them to meet modern standards, making these integrations accessible to everyone.
+
+Characteristics:
+- Complex setup process; only configurable via YAML, without UI-based setup.
+- May lack active code ownership and maintenance.
+- Could be missing recent updates or bug fixes.
+- Documentation may still be aimed at developers.
+
+#### 📦 Custom
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:122]:
```suggestion
#### 📦 Custom

```

File: adr/0022-integration-quality-scale.md [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+
+Characteristics:
+- Not included in the official Home Assistant releases.
+- Manually installable or installable via community tools, like HACS.
+- Maintained by individual developers or community members.
+- User experience may vary widely.
+- Functionality, security, and stability can vary widely.
+- Documentation may be limited.
+
+### Adjusting the tier of an integration
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:137]:
```suggestion
### Adjusting the tier of an integration

```

File: adr/0022-integration-quality-scale.md [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+Home Assistant encourages our contributors to get their integrations to the highest possible tier, to provide an excellent coding experience for our contributors and the best experience for our users.
+
+When an integration reaches the minimum requirements for a certain tier, a contributor can open a pull request to adjust the scale for the integration. This request needs to be accompanied by the full checklist for each rule of scale (including all rules of lower tiers), demonstrating that it has met those requirements.
+
+Once the Home Assistant core team reviews and approves it, the integration will display the new tier as of the next major release of Home Assistant.
+
+Besides upgrading an integration to a higher tier on the scale, it is also possible for an integration to be downgraded to a lower tier. This can, for example, happen when there is no longer an active integration code owner. In this specific example, the integration will be downgraded to “Bronze”, even if it otherwise fully complies with the “Platinum” tier.
+
+### Adjustments to rules contained in each tier
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:146]:
```suggestion
### Adjustments to rules contained in each tier

```

**Review by frenck — CHANGES_REQUESTED** [review]
Not sure about the consequence part of the ADR, I think we should link to the developer documentation from here as well.

File: adr/0022-integration-quality-scale.md [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+
+If rules to our quality integration scales have been adjusted, it is possible that an integration with the Works with Home Assistant program doesn’t comply with the newly adjusted standards yet. At that moment, the existing “Gold” tier is grandfathered. Once a new or extra device is certified for the same integration, the integration must be updated to also comply with the new or adjusted standards.
+
+When all devices are removed from the Works with Home Assistant program of a grandfathered integration, the grandfathered flag will also be released and the integration quality scale tier adjusted to the current status of the integration.
+
+## Consequences
+
+The integration quality scale will:
+- give a clear and transparent framework for contributors to understand the quality expectations for their integrations.
+- put pressure on the maintenance of the developer documentation to make sure the documentation is up-to-date.
```
Comment (joostlek) [inline adr/0022-integration-quality-scale.md:173]:
```suggestion
- put pressure on the maintenance of the developer documentation to make sure the documentation is up-to-date. These can be found [here](https://developers.home-assistant.io/docs/core/integration-quality-scale/)
```

File: adr/0022-integration-quality-scale.md [hunk 21]
```diff
@@ -0,0 +1,173 @@
+# 22. Integration quality scale
+
+Date: yyyy-mm-dd
```
Comment (frenck) [inline adr/0022-integration-quality-scale.md:3]:
```suggestion
Date: 2024-11-20
```

**Review by frenck — APPROVED** [review]

**Pull request merged**